### PR TITLE
Reduce cxxtools::Json footprint

### PR DESCRIPTION
**Pull request by sinanyavas-eaton**

### CODE REVIEW

File: ecpp/assets_in.ecpp [hunk 1]
```diff
@@ -237,9 +238,11 @@ bool database_ready;
       }
       persist::export_asset_json (reply.out (), &listElements);
     } else {
-      cxxtools::JsonSerializer serializer(reply.out ());
-      serializer.beautify(true);
-      serializer.serialize(assets).finish();
+      for(const auto& i : assets) {
```
Comment (ericclappier-eaton) [inline ecpp/assets_in.ecpp:240]: beautify = true

**Review by ericclappier-eaton — DISMISSED** [review]

File: ecpp/alert_templates.ecpp [hunk 2]
```diff
@@ -112,9 +111,8 @@ while(template_name){
     cxxtools::SerializationInfo &si_templat_obj= si.addMember("");
     si_templat_obj.addMember("name") <<= std::string(template_name);
     cxxtools::SerializationInfo &si_templat = si_templat_obj.addMember("template");
-    std::stringstream input (templat.get());
-    cxxtools::JsonDeserializer  deserializer(input);
-    deserializer.deserialize(si_templat);
+    std::istringstream input (templat.get());
+    JSON::readFromStream(input, si_templat);
```
Comment (FrancoisRegisDegott-eaton) [inline ecpp/alert_templates.ecpp]: JSON::readFromString(templat.get(), si_templat);

File: ecpp/conf_scan.ecpp [hunk 3]
```diff
@@ -27,8 +27,8 @@
 #include <fty_common_macros.h>
 #include <fty_common_mlm_pool.h>
 #include <fty_common_rest_helpers.h>
-#include <cxxtools/jsondeserializer.h>
 #include <fty_common_rest_audit_log.h>
+#include <fty_common_json.h>
 
 static zmsg_t* req_setconf (zuuid_t *uuid, std::string json)
```
Comment (FrancoisRegisDegott-eaton) [inline ecpp/conf_scan.ecpp]: const std::string& json

File: ecpp/current.ecpp [hunk 4]
```diff
@@ -344,10 +344,5 @@ bool database_ready;
     }//for
 
     // serialize json response
-    std::ostringstream response(std::ios_base::out);
-    cxxtools::JsonSerializer serializer(response);
-    serializer.inputUtf8(true);
-    serializer.serialize(siRoot).finish();
-
-    reply.out() << response.str();
+    reply.out() << JSON::writeToString(siRoot);
```
Comment (FrancoisRegisDegott-eaton) [inline ecpp/current.ecpp]: true (explicitize beautify)

**Review by FrancoisRegisDegott-eaton — DISMISSED** [review]

**Review by FrancoisRegisDegott-eaton — DISMISSED** [review]

File: ecpp/sysinfo.ecpp [hunk 5]
```diff
@@ -382,10 +380,8 @@ sysinfo_detail (
             in.close();
             std::string bios_release (tmp.str ());
 
-            std::stringstream input (bios_release, std::ios_base::in);
-            cxxtools::JsonDeserializer deserializer (input);
             cxxtools::SerializationInfo si;
```
Comment (FrancoisRegisDegott-eaton) [inline ecpp/sysinfo.ecpp:383]: we could have simplified here... with a simple readFromFile() call

**Review by FrancoisRegisDegott-eaton — APPROVED** [review]

File: ecpp/assets_in.ecpp (hunk 1 above)
Comment (FrancoisRegisDegott-eaton) [reply ecpp/assets_in.ecpp:240]: done

**Pull request merged**